src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -625,6 +626,45 @@ public ResponseEntity<Void> recreateBuildPlans(@PathVariable Long exerciseId) {
         return ResponseEntity.ok().build();
     }
 
+    /**
+     * Consolidates the reset functionality for a programming exercise by performing a set of operations
+     * as specified in the ProgrammingExerciseResetOptionsDTO for an exercise given an exerciseId.
+     *
+     * The available operations include:
+     * 1. deleteBuildPlans: Deleting all student build plans (except BASE/SOLUTION).
+     * 2. deleteRepositories: Deleting all student repositories (requires: 1. deleteBuildPlans == true).
+     * 3. deleteParticipationsSubmissionsAndResults: Deleting all participations, submissions, and results.
+     * 4. recreateBuildPlans: Deleting and recreating the BASE and SOLUTION build plans.
+     *
+     * @param exerciseId                         - Id of the programming exercise to reset.
+     * @param programmingExerciseResetOptionsDTO - Data Transfer Object specifying which operations to perform during the exercise reset.
+     * @return ResponseEntity<Void> - The ResponseEntity with status 200 (OK) if the reset was successful.
+     */
+    @PutMapping(RESET)
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<Void> reset(@PathVariable Long exerciseId, @RequestBody ProgrammingExerciseResetOptionsDTO programmingExerciseResetOptionsDTO) {
```

Comment:
Since we have this new Endpoint, is it possible to delete the others that were only meant to reset certain things?

Reply:
It seems that I can safely remove exercise's `/cleanup` but exercise's `/reset` is used by quiz exercises. I will remove `/cleanup` since it seems anyways to be located in the wrong file.